test/functional/wallet_miniscript_decaying_multisig_descriptor_psbt.py

```diff
@@ -0,0 +1,177 @@
+            all(address == change_addresses[0] for address in change_addresses)
+
+        self.log.info("Get a mature utxo to send to the multisig...")
+        coordinator_wallet = participants["signers"][0]
+        self.generatetoaddress(self.nodes[0], 101, coordinator_wallet.getnewaddress())
+
+        deposit_amount = 6.15
+        multisig_receiving_address = participants["multisigs"][0].getnewaddress()
+        self.log.info("Send funds to the resulting multisig receiving address...")
+        coordinator_wallet.sendtoaddress(multisig_receiving_address, deposit_amount)
+        self.generate(self.nodes[0], 1)
+        for participant in participants["multisigs"]:
+            assert_approx(participant.getbalance(), deposit_amount, vspan=0.001)
+
+        self.log.info("Send a transaction from the multisig! We can always spend with all signers!")
+        to = participants["signers"][self.N - 1].getnewaddress()
+        value = 1.5
+        self.log.info("First, make a sending transaction, created using `walletcreatefundedpsbt` (anyone can initiate this)...")
+        psbt = participants["multisigs"][0].walletcreatefundedpsbt(inputs=[], outputs={to: value}, feeRate=0.00010)
+
+        psbts = []
+        self.log.info("Now at least M users check the psbt with decodepsbt and (if OK) signs it with walletprocesspsbt...")
+        for m in random.sample(range(self.M), self.M):
+            signers_multisig = participants["multisigs"][m]
+            self._check_psbt(psbt["psbt"], to, value, signers_multisig)
+            signing_wallet = participants["signers"][m]
+            partially_signed_psbt = signing_wallet.walletprocesspsbt(psbt["psbt"])
+            psbts.append(partially_signed_psbt["psbt"])
+
+        self.log.info("Finally, collect the signed PSBTs with combinepsbt, finalizepsbt, then broadcast the resulting transaction...")
+        combined = coordinator_wallet.combinepsbt(psbts)
+        finalized = coordinator_wallet.finalizepsbt(combined)
+        coordinator_wallet.sendrawtransaction(finalized["hex"])
+
+        self.log.info("Check that balances are correct after the transaction has been included in a block.")
+        self.generate(self.nodes[0], 1)
+        assert_approx(participants["multisigs"][0].getbalance(), deposit_amount - value, vspan=0.001)
+        assert_equal(participants["signers"][self.N - 1].getbalance(), value)
+
+        self.log.info("Send more transactions from the multisig as required signers decay, this time with a daisy chained signing flow (one after another in series)!")
+        sent = value
+        for locktime in self.locktimes:
+            self.M -= 1
+            self.log.info(f"At block height >= {locktime} this multisig is {self.M}-of-{self.N}")
+            current_height = self.nodes[0].getblock(self.nodes[0].getbestblockhash())['height']
+
+            psbt = participants["multisigs"][0].walletcreatefundedpsbt(inputs=[], outputs={to: value}, feeRate=0.00010, locktime=locktime)
```

Comment:
An interesting thing I learned with this test was psbt behavior related to `locktime`. If I created a psbt at a blockheight of 800,000 when multisig was 4-of-4 (maybe I want to pre-sign it with a cold wallet and save it for later), and then I wanted to spend it when the multisig decayed to 3-of-4 at blockheight 840,000, then wallet would not process that psbt as `complete` with 3 valid signatures like I expected (bc I guess it's using locktime of when psbt was created to know how to interpret the Miniscript and not current blockheight)

I didn't investigate further or learn why that is.. or if there is a fundamental reason for this behavior or if `walletprocesspsbt` can/should be improved

Reply:
> then wallet would not process that psbt as complete with 3 valid signatures like I expected

Do you mean the 4-signature transaction was rejected?

Reply:
I just got confused here. CLTV opcode checks against spending transaction `nLockTime`. So behavior is exactly as expected

> Do you mean the 4-signature transaction was rejected?

No I didn't mean this. 4-signature transaction would never be rejected. But I explained this confusingly so you can ignore my comment above